File: src/cmd/window.ts
import { TauriCmd } from "@/enums";

import { invokeWrapper } from ".";

export type OpenModalWindowArgs = {
  path: string;
  title?: string;
  width?: number;
  height?: number;
  label?: string;
};

export const openModalWindow = (args: OpenModalWindowArgs) =>
  invokeWrapper<string>(TauriCmd.OpenModalWindow, { ...args });

export const closeModalWindow = (label: string) =>
  invokeWrapper<void>(TauriCmd.CloseModalWindow, { label });

export const notifyModalWindowReady = (label: string) =>
  invokeWrapper<void>(TauriCmd.ModalWindowReady, { label });

/** 主窗空闲时后台预热隐藏 modal Webview（不显示、不触发蒙层） */
export const preloadModalWindow = () => invokeWrapper<void>(TauriCmd.PreloadModalWindow);

File: src-tauri/src/cmd/window.rs
use tauri::AppHandle;

use crate::utils::window as window_util;

#[tauri::command]
pub async fn open_modal_window(
    app: AppHandle,
    path: String,
    title: Option<String>,
    width: Option<f64>,
    height: Option<f64>,
    label: Option<String>,
) -> Result<String, String> {
    let label = window_util::open_modal_window(&app, path, title, width, height, label)?;
    crate::log_info!("cmd.window.open_modal_window ok label={label}");
    Ok(label)
}

#[tauri::command]
pub async fn close_modal_window(app: AppHandle, label: String) -> Result<(), String> {
    window_util::close_modal_window(&app, &label)?;
    crate::log_info!("cmd.window.close_modal_window ok label={label}");
    Ok(())
}

#[tauri::command]
pub async fn modal_window_ready(app: AppHandle, label: String) -> Result<(), String> {
    window_util::modal_window_ready(&app, &label)?;
    crate::log_info!("cmd.window.modal_window_ready ok label={label}");
    Ok(())
}

#[tauri::command]
pub async fn preload_modal_window(app: AppHandle) -> Result<(), String> {
    window_util::preload_modal_window(&app)?;
    crate::log_info!("cmd.window.preload_modal_window ok");
    Ok(())
}
